server/repositories/entities/reportDefinition.ts

```diff
@@ -0,0 +1,17 @@
+import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from 'typeorm'
+import ReportDefinitionFields from './reportDefinitionFields'
+
+@Entity('report_definition')
+export default class ReportDefinition {
+  @PrimaryGeneratedColumn()
+  id: number
+
+  @Column()
+  type: string
+
+  @Column()
+  version: number
+
+  @OneToMany(() => ReportDefinitionFields, entity => entity.reportDefinition, { eager: true })
```

Comment:
I think you might be able to do away with the `ReportDefinitionFields` entity using a @ManyToMany and a @JoinTable https://github.com/typeorm/typeorm/blob/master/docs/many-to-many-relations.md

Reply:
Well spotted, updated this now to use ManyToMany